sql/plan/alter_table.go

```diff
@@ -722,6 +723,28 @@ func (i *dropColumnIter) Next(ctx *sql.Context) (sql.Row, error) {
 		return nil, err
 	}
 
+	// drop constraints that only reference the dropped column
+	cat, ok := i.alterable.(sql.CheckAlterableTable)
+	if !ok {
+		return nil, errors.New("failed to cast to CheckAlterableTable")
+	}
```

Comment:
(minor) Probably better to not return an error here... we want to let people extend GMS with their own implementations, and ideally, they shouldn't have to implement all interfaces in order to get something working. So, it would be nice if this code checked if the table has the capability to alter checks, and if so, it removes them, but otherwise it just drops the column.